### ISSUE TITLE
Config test fails if home dir has .eslintrc file

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#6559]
While i was working on `master` [this config unit test](https://github.com/eslint/eslint/blob/master/tests/lib/config.js#L265) fails if you have a `.eslintrc` file inside your home directory. I think this test was introduced with #6538.

To reproduce just create a `.eslintrc` file inside your HOME directory and then run the tests.

```js
{
	"env" : {
		"es6" : true
	}
}
```

output:

```sh
 11186 passing (48s)
  1 failing

  1) Config getConfig() should throw an error when an invalid path is given:
     AssertionError: expected [Function] to throw an error
      at Function.assert.throws (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:1120:52)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:271:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:326:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:319:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:528:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)
```

Note: This issue is similar to what was happening here #6206

**Comment by platinumazure** [comment]
Ack, probably needs a mock filesystem setup in that test.

I'm happy to look at this but I will probably need some guidance.

**Comment by gyandeeps** [comment]
no worries @platinumazure . Let me know if you need more help since i did the work on mock before.

**Comment by platinumazure** [comment]
I'll work on this tonight (i.e., soon) and have either a PR or an update here soon.

**Comment by platinumazure** [comment]
@gyandeeps Any chance you could try my branch (see PR #6566) and see if that test passes for you? Thanks!
